1.20 adding charms in charm slots (Trinkets "fix")

Shibva opened the pull request:
Add hand/charm and moved 	``divaCharm`` and ``goddessCharm`` to section (hopefuly I did this right)

Creates a new charm slot in the mainland section 

@williewillus NOTE: needs to be run like you said; I'm still fairly new to all of these things so I'm not sure how to go about running it. 

I also want to give the trinket case the needed trinket compatibility since it is missing as Curios has it but it is outside of my ability to implement such

Hopefully this will generate the other files as needed

Shibva committed:
Update FabricItemTagProvider.java

Add hand/charm and moved 	``divaCharm`` and ``goddessCharm`` to section (hopefuly I did this right)

Shibva commented:
also needs backport to 1.19.2 as a reminder :D

williewillus commented:
You need to run `./gradlew :Fabric:runFabricDatagen` (`./gradlew.bat :Fabric:runFabricDatagen` on windows) and commit the changes it generates.

williewillus commented:
Also, you're missing this change from the other PR. Is that intentional? https://github.com/VazkiiMods/Botania/pull/4443/files#diff-478bd022a79d7cb1ddd31a5f0a03f5d5ca5446fafcfe9f4abf6fc5b7e06f20f8

Shibva committed:
Finalized chages

- runFabricDatagen ran wtih previous commit made above
  (NOTE:  Error message generated upon run: `com.mojang.authlib.exceptions.MinecraftClientHttpException: Status: 401`
- Added `"hand/charm"`  slot; hopefully this dosen't create a second slot with other mods that add this slot in.
- Cached Data changes from datagen run (unsure if this is needed but including just in case)

Reviews needed before approval

Shibva commented:
Please review this before approving it; I'll not lie when I say I only have a fundamental understanding of how the datapack aspect of this works. I will admit I do not know how to code with Java but I feel that my understanding of Minecraft's datapack system that is integrated with mods is solid enough for me to contribute in those areas.

I would also work in Trinket Case compact with trinkets but that is outside of my range of possibility

williewillus commented:
The code looks good to me. Did you run the game and see if it worked ingame? If so I can land this.

Shibva commented:
game boots but its not creatin the charm slot for some reason

Shibva commented:
Ok I see what needs to be done now

I'll need to add a custom slot in the same place entities are located

I'll finish that up tomorrow when I have the chance

Shibva commented:
Thankfully, in the source, there's a test mod that shows me what to do

Shibva committed:
Create charm.json

finishes the implementation

Thank You TheWinABagel for the file I copied over to finish the job

Shibva commented:
Ok, one look at another source later and a quick grab to make things quicker and its done.

ran it again and it now all works with the slot added and everything @williewillus

williewillus commented:
Hm. It doesn't seem right that we have jsons in trinkets' namespace. Let me check the docs about how this is supposed to work.

Shibva commented:
It should be the case; based on another mod I looked at with open source, it appears to be in the right spot

unilock commented:
It's important to mention that the "charm" slot *does not exist* in the Trinkets API by default, although the texture for it does.

As such, the tooltip for any item that can be put in that slot will be missing a translation key; thus the following entry should be added to the lang file(s):
```
"trinkets.slot.hand.charm": "Charm"
```

For what it's worth, it seems that the most popular mod that implements a charm slot, "[Charm of Undying](https://github.com/illusivesoulworks/charmofundying)", defines the slot as `"charm/charm"`, not `"hand/charm"`. ([source](https://github.com/illusivesoulworks/charmofundying/blob/0e1f38d751b2915acb557c45c74b1c52f7d24827/fabric/src/main/resources/data/trinkets/slots/charm/charm.json))

Shibva commented:
That is true; this is more or less a fallback in case non-such are present; the code should work, and I have tested it in the space.

Shibva commented:
I was told to make this to be applied for 1.20 and then retroactively for 1.19.2

Shibva commented:
Feel free to make the changes for it to work then as well, this was more of a nitpick fix for trinkets mod

Shibva commented:
Renamed this; this is probably why it has seen no movement lel
